ffi/driver.cc

```diff
@@ -36,6 +38,17 @@ void init_ffi_driver(py::module_ &m) {
         .def("sync", &Driver::sync)
         .def("collect_returns", &Driver::collectReturns)
         .def("time", &Driver::time, "rounds"_a = 10, "warmpups"_a = 3);
+
+    // Serialization
+    m.def("load_target", &loadTarget);
+    m.def("load_device", &loadDevice);
+    m.def("load_array", &loadArray);
+    m.def("dump_target", &dumpTarget, "target"_a);
+    m.def("dump_device", &dumpDevice, "device"_a);
+    m.def("dump_array", &dumpArray, "array"_a);
```

Comment:
If you are using bytes (see my comment for `dumpArray`), PyBind11 requires special wrapper code (https://pybind11.readthedocs.io/en/stable/advanced/cast/strings.html#return-c-strings-without-conversion)